### PR TITLE
Expose XMLHttpRequest.abort() method on Promise

### LINKED ISSUE #24: Abort an ajax request
Hi 

jQuery.ajax initally returns the jqXHR object, on which we can call the abort method.
Is there a way to abort an ajax request with ic-ajax?

### PR DESCRIPTION
**Pull request by Turbo87**
as suggested in https://github.com/tildeio/rsvp.js/issues/394 by @stefanpenner 

resolves #24

### CODE REVIEW

**Comment by stefanpenner** [comment]
I proposed it as a potential work-around.

> although I am dubious about the exposing the xhr.

I do not believe this is quite what will work though.

**Comment by Turbo87** [comment]
could you elaborate on that? what else could be done to expose the abort method?

**Comment by stefanpenner** [comment]
problems:

```
promise = ajax().then()
promise.abort === undefined
```
- does abort cancel?
- what does cancel mean
- how downstream subscribers notified?
- what happens if it already succeeded.

Until the promise spec decides on a cancellation strategy I am dubious of exposing it here.

Solution, use a task instead. 

``` js
var task = ajaxTask(...)
task.cancel()
task.promise;
```

No need to confuse people, as not all promises are abortable.

**Comment by Turbo87** [comment]
are you talking about [Task.js](http://taskjs.org/)? how can that be integrated with ic-ajax? I fail to understand how that should work without modifying ic-ajax.

**Comment by stefanpenner** [comment]
TL;DR promise cancelation is fraught with peril, some context -> https://github.com/promises-aplus/cancellation-spec/issues. This is the reason for my apprehension in shoe horning something in here.

The recommendation of a task, is not to use task.js but to consider an API where a touple of promise + cancel handler are returned. This skips the issues of cancelable promises, by virtue of introducing an additional concept.

Another thing to consider is the html5 fetch api.
